src/opm/input/eclipse/Schedule/KeywordHandlers.cpp

```diff
@@ -766,19 +766,48 @@ File {} line {}.)", wname, location.keyword, location.filename, location.lineno)
     }
 
     void Schedule::handleGRUPNET(HandlerContext& handlerContext) {
-        for (const auto& record : handlerContext.keyword) {
-            const auto& groupName = record.getItem("NAME").getTrimmedString(0);
-
-            if (!this->snapshots.back().groups.has(groupName))
-                addGroup(groupName , handlerContext.currentStep);
 
-            int table = record.getItem("VFP_TABLE").get< int >(0);
+        auto network = this->snapshots.back().network.get();
+        std::vector<Network::Node> nodes;
+        for (const auto& record : handlerContext.keyword) {
+            const std::string& groupNamePattern = record.getItem<ParserKeywords::GRUPNET::NAME>().getTrimmedString(0);
+            const auto group_names = this->groupNames(groupNamePattern);
+            if (group_names.empty())
+                this->invalidNamePattern(groupNamePattern, handlerContext);
+            const auto& pressure_item = record.getItem<ParserKeywords::GRUPNET::TERMINAL_PRESSURE>();
+            const int vfp_table = record.getItem<ParserKeywords::GRUPNET::VFP_TABLE>().get<int>(0);
 
-            auto new_group = this->snapshots.back().groups.get( groupName );
-            if (new_group.updateNetVFPTable(table))
-                this->snapshots.back().groups.update( std::move(new_group) );
+            for (const auto& group_name : group_names) {
+                auto& group = this->snapshots.back().groups.get(group_name);
+                group.updateNetVFPTable(vfp_table);
```

Comment:
By handling GRUPNET related in the Network class, we do not need to use facility of the `Group` any more.  So it is not necessary to `updateNetVFPTable(vfp_table)` here and the `group` above should be `const` since we should not change things related to it. 

Furthermore, with the PR, we **might** be able to remove the `vfp_table` member of the `Group` class and related member functions.

Reply:
Done except for removing `vfp_table` member of the `Group` class and members. I propose to do such clean up in a separate PR